fsw/cfe-core/src/tbl/cfe_tbl_api.c

```diff
@@ -761,19 +766,157 @@ int32 CFE_TBL_Load( CFE_TBL_Handle_t TblHandle,
                     /* Determine whether the load is to occur from a file or from a block of memory */
                     if (SrcType == CFE_TBL_SRC_FILE)
                     {
-                        /* Load the data from the file into the specified buffer */
-                        Status = CFE_TBL_LoadFromFile(WorkingBufferPtr, RegRecPtr, (const char *)SrcDataPtr);
+                        CFE_FS_Header_t      StdFileHeader;
+                        CFE_TBL_File_Hdr_t   TblFileHeader;
+                        int32                FileDescriptor;
+                        const char *         Filename = SrcDataPtr;
+                        size_t               FilenameLen = strlen(Filename);
+                        uint32               NumBytes;
+                        uint8                ExtraByte;
+
```

Comment:
This is against our code standard (declare variables up front).

Curious about what the CCB thinks about this approach.  Coding standard suggest shorter functions if I recall correctly, vs 1 big function.

Reply:
It's 363 lines, far shorter than CFE_TBL_Register() [450 lines] and similar to CFE_SB_SendMsgFull() [321 lines]...It is possible that it will be split out again when merging the code with CFE_TBL_LoadCmd() (see #600 ). But I wanted to simplify code and tests to address this ticket, then approach LoadCmd() separately.

If there's objection, I can move back to using an external function. HOWEVER, I definitely do not like the logic in Load() and would keep this simpler, more direct logic.

The old code calls LoadFromFile() and checks the return code ( https://github.com/nasa/cFE/blob/60a5f6580d1e737f46efce0418aba5b8dac6df6a/fsw/cfe-core/src/tbl/cfe_tbl_api.c#L753 ) and generate a SysLog event sometimes, then at the end of Load() it generates some sort of event ( https://github.com/nasa/cFE/blob/60a5f6580d1e737f46efce0418aba5b8dac6df6a/fsw/cfe-core/src/tbl/cfe_tbl_api.c#L896 ). So event generation is about as far away from error detection as possible. :o

I'd like to also discuss with the CCB the validity of functional testing code that pokes at internal API's. We do it everywhere (CFE_SB_SendMsgFull(), for example) but it seems like not true black-box testing. I'd rather change the tbl_UT to only call CFE_TBL_Load() and not whatever internal function is behind it.

Reply:
CFE_TBL_Register is a cyclomatic complexity of 49 CFE_SB_SendMsgFull is 32.  The recommendation for flight code is 10 (https://swehb.nasa.gov/display/SWEHBVC/SWE-134+-+Safety+Critical+Software+Design+Requirements).  Any new/refactored code should keep this number as a goal, and avoid raising complexity where possible.

Simpler logic is definitely preferred.

Note tbl_UT primary goal is white box testing (coverage and unit function).  Functional/black box/requirements verification is through the build verification tests and test apps (black box).

Reply:
Copy all, I'll close this pull request and refactor and resubmit.